### PR TITLE
Search results drill down

### PR DESCRIPTION
**Pull request by stevenwilkin**
Needs rebased when #238 lands to remove ccae4cc.

Search results:

image

Links through to:

image

### CODE REVIEW

File: spec/controllers/locations_controller_spec.rb [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
+
+    before do
+      allow(Locations).to receive(:find).with(invalid_id)
+      allow(Locations).to receive(:find).with(valid_id).and_return(location)
+    end
+
+    specify 'with an invalid id' do
+      get :show, id: invalid_id
+
+      expect(response).to redirect_to(appointments_path)
```
Comment (gjvis) [inline spec/controllers/locations_controller_spec.rb]: why not simply 404? I wonder if there are any implications to doing this in terms of caching etc. I also wonder what others on the team think in terms of usability

**Comment by gjvis** [comment]
Looks great, one small question around invalid id behaviour...

File: spec/controllers/locations_controller_spec.rb (hunk 1 above)
Comment (stevenwilkin) [inline spec/controllers/locations_controller_spec.rb]: Good spot, the error scenarios in `LocationsController#index` redirect and I'd blindly done the same thing here

**Comment by stevenwilkin** [comment]
Rebased. Renders standard 404 when requesting an invalid location id

**Comment by gjvis** [comment]
Looks good :+1:

File: features/pages/location_page.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module Pages
+  class Location < SitePrism::Page
+    set_url '/locations/{id}'
```
Comment (andrewgarner) [inline features/pages/location_page.rb]: Is this a conscious decision to include the resource type in the URL?

We haven't followed that convention with guides for example.

Comment (stevenwilkin) [inline features/pages/location_page.rb]: No. This is building on what is [already in place with `Pages::Locations`](https://github.com/guidance-guarantee-programme/pension_guidance/blob/37fae3cf14fdeb87098c8fb270efe1e0704c0ab4/features/pages/locations_page.rb#L3).

Comment (gjvis) [inline features/pages/location_page.rb]: It is a conscious decision yes. It was defined as a `resources` in the routes file to facilitate exactly this.

**Comment by stevenwilkin** [comment]
Reminder to remove a90ed906b8cfb2851a282d3f6635c00003f76257 and d23fb3be8f0cc9114ac88750a72447e7c1ecaecb when the relevant PRs are merged.

**Pull request merged**